### PR TITLE
Bucket policies parser should use minio-go/pkg/policy instead.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Bucket policies parser should use minio-go/pkg/policy instead.

## Motivation and Context
Removing redundant parser logic.

## How Has This Been Tested?
Manually, using `go test`.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/bucket-policy-parser.go [hunk 1]
```diff
@@ -26,9 +26,12 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/minio/minio-go/pkg/policy"
 	"github.com/minio/minio-go/pkg/set"
 )
 
+var sentinelBucketPolicy = policy.BucketAccessPolicy{}
```
Comment (donatello) [inline cmd/bucket-policy-parser.go]: Perhaps `emptyBucketPolicy` is a more straightforward name for this? Sentinels are usually used in the context of sorting as a sort of zero/border value.

**Review by donatello — DISMISSED** [review]
Looks like a straightforward change. LGTM - have a comment about renaming an empty policy.

File: cmd/bucket-policy-parser.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/bucket-policy-parser.go]: I can do that @donatello - i used sentinel in context of default value.

**Review by krishnasrinivas — APPROVED** [review]

File: cmd/bucket-policy-parser.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/bucket-policy-parser.go]: @donatello please review again.

**Review by donatello — APPROVED** [review]

**Pull request merged**